app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,34 @@ def show
     @post = @user.posts.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.new(author_id: params[:user_id], post_id: @post.id)
+    @like.save
+    redirect_to user_post_path
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counters = 0
```

Comment:
- [ ] You have a typo here that causes an error when I try to create a post. This should be `likes_counter` and not `likes_counters`.
![image](https://github.com/Microverse-Fullstack-Program/rails-blog-app/assets/45460192/18a8f2b2-f0dd-464f-a680-6235306c8479)